actions/views/channel_sidebar.ts

```diff
@@ -93,10 +97,73 @@ export function adjustTargetIndexForMove(state: GlobalState, categoryId: string,
     // the channel being removed. For example, if we're moving channelA from [channelA, channelB, channelC] to
     // [channelB, channelA, channelC], newIndex would currently be 2 (which comes after channelB), but we need
     // it to be 1 (which comes after channelB once channelA is removed).
-    const sourceIndex = category.channel_ids.indexOf(channelId);
+    const sourceIndex = category.channel_ids.indexOf(channelIds[0]);
     if (sourceIndex !== -1 && sourceIndex < newIndex) {
         newIndex -= 1;
     }
 
-    return newIndex;
+    return Math.max(newIndex - removedChannelsAboveInsert.length, 0);
+}
+
+export function clearChannelSelection() {
+    return (dispatch: DispatchFunc) => {
```

Comment:
Can this just return the action without dispatching it or does that run into some weird typing issue?

Reply:
Weird typing issue, because it expects the object returned to be of type: `{type: string, data: any}`
Since there's no `data` here, I have to wrap it.

Reply:
Strange. I'm surprised undefined doesn't satisfy any. I guess it just doesn't satisfy it implicitly.